applications/mne_analyze/extensions/dipolefit/dipolefit.cpp

```diff
@@ -140,13 +218,275 @@ QWidget *DipoleFit::getView()
+
+        norm = sqrt(pow(tempEcd.Q(0),2)+pow(tempEcd.Q(1),2)+pow(tempEcd.Q(2),2));
+
+        to.setX(tempEcd.Q(0)/norm);
+        to.setY(tempEcd.Q(1)/norm);
+        to.setZ(tempEcd.Q(2)/norm);
+
+//        qDebug()<<"EcdDataTreeItem::plotDipoles - from" << from;
+//        qDebug()<<"EcdDataTreeItem::plotDipoles - to" << to;
+
+        QQuaternion final = QQuaternion::rotationTo(from, to);
+
+        //Set dipole position and orientation
+        QMatrix4x4 m;
+        m.translate(pos);
+        m.rotate(final);
+
+        Qt3DCore::QTransform *pTransform = new Qt3DCore::QTransform();
+        pTransform->setMatrix(m);
+
+        Qt3DExtras::QPhongMaterial *pMaterial = new Qt3DExtras::QPhongMaterial();
+        pMaterial->setAmbient(QColor(rand()%255, rand()%255, rand()%255));
+
+        Qt3DExtras::QConeMesh *pDipolGeometry = new Qt3DExtras::QConeMesh();
+        pDipolGeometry->setBottomRadius(0.001f);
+        pDipolGeometry->setLength(0.003f);
+
+        Qt3DCore::QEntity *pEntity = new QEntity(pRootEntity.data());
+
+        pEntity->addComponent(pTransform);
+        pEntity->addComponent(pMaterial);
+        pEntity->addComponent(pDipolGeometry);
+    }
+
+    return pRootEntity;
+}
+
+
+//*************************************************************************************************************
+
+void DipoleFit::onBrowseButtonClicked()
+{
+    //Get the path
+    QString filePath = QFileDialog::getOpenFileName(m_pDipoleFitControl,
+                                tr("Open File"),
+                                QDir::currentPath()+"/MNE-sample-data",
+                                tr("fif File(*.fif)"));
+    if(!filePath.isNull()) {
+        m_dipoleSettings.setMeasurementFilePath(filePath);
+
+    }
+}
+
+
+//*************************************************************************************************************
+
+void DipoleFit::onFitButtonClicked()
+{
+    QString sFitName = m_pDipoleFitControl->getFitName();
+    QString sModelPath = ECD_SET_MODEL_DEFAULT_DIR_PATH + sFitName;
+    if(sFitName.isEmpty()) {
+        m_pCommu->publishEvent(STATUS_BAR_MSG, QStringLiteral("Fit name invalid"));
+        return;
+    }
+
+    m_analyzeData->loadEcdSetModel(m_dipoleSettings.getSettings(), sModelPath);
+}
+
+
+//*************************************************************************************************************
+
+void DipoleFit::onActiveModelSelected(const QString &sModelName)
+{
+    if(m_bInitFinished == false) {
+        return;
+    }
+
+    //find selected model
+    auto result = std::find_if(m_vEcdSetModels.begin(), m_vEcdSetModels.end(),
+                 [sModelName]( const QPair<QSharedPointer<EcdSetModel>, QSharedPointer<QEntity>> &modelPair) {
+        return modelPair.first->getModelName() == sModelName;
+    });
+
+    if(result != m_vEcdSetModels.end()) {
+        //change visible 3D-model
+        bool removeSuccessful = m_pDisplayModel->removeEntityTree(m_pDipoleRoot);
```

Comment:
If this only concerns visibility, you might want to consider the setEnabled-method. On the other hand, this will reveal potential flaws in the display system, so why not leave it :smile:

Reply:
Yes u are probably right. I left a TODO comment.